java/ql/src/Security/CWE/CWE-489/DebuggableAttributeEnabled.ql

```diff
@@ -13,9 +13,8 @@
 import java
 import semmle.code.xml.AndroidManifest
 
-from AndroidXmlAttribute androidXmlAttr
+from AndroidApplicationXmlElement androidAppElem
 where
-  androidXmlAttr.getName() = "debuggable" and
-  androidXmlAttr.getValue() = "true" and
-  not androidXmlAttr.getLocation().getFile().getRelativePath().matches("%build%")
-select androidXmlAttr, "The 'android:debuggable' attribute is enabled."
+  androidAppElem.isDebuggable() and
+  not androidAppElem.getFile().isInBuildDirectory()
```

Comment:
As mentioned above, you can just use [casting](https://codeql.github.com/docs/ql-language-reference/expressions/#casts) to access specific `AndroidManifestXmlFile` predicates, since the file returned by `getFile` is already of that class:

```suggestion
  not androidAppElem.getFile().(AndroidManifestXmlFile).isInBuildDirectory()
```

Reply:
Updated in commit [bee310d](https://github.com/github/codeql/pull/9939/commits/bee310d9e0c0e8aa9927572d28dad4bd5c414981).